packages/trimerge-sync/src/TrimergeClient.ts

```diff
@@ -355,6 +391,8 @@ export class TrimergeClient<
   private addCommit(
     commit: Commit<CommitMetadata, Delta>,
     type: AddCommitType,
+    resolve?: () => void,
+    reject?: (e: unknown) => void,
```

Comment:
Should we require that both require and reject are defined if either is defined? Like `resolveReject?: { resolve: () => void, reject: (e: unknown) => void }`